pkg/apitype/policy.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apitype
+
+// CreatePolicyPackRequest defines the request body for creating a new Policy
+// Pack for an organization. The request contains the metadata related to the
+// Policy Pack.
+type CreatePolicyPackRequest struct {
+	Name        string   `json:"name"`
+	Version     int      `json:"version"`
+	DisplayName string   `json:"displayName"`
+	Policies    []Policy `json:"policies"`
+}
+
+// CreatePolicyPackResponse is the response from creating a Policy Pack. It returns
+// a URI to upload the Policy Pack zip file to.
+type CreatePolicyPackResponse struct {
+	UploadURI string `json:"uploadURI"`
+}
+
+// RequiredPolicy is the information regarding a particular Policy that is required
+// by an organization.
+type RequiredPolicy struct {
+	Name         string `json:"name"`
+	Version      int    `json:"version"`
+	DisplayName  string `json:"displayName"`
+	PackLocation string `json:"packLocation"`
+}
+
+// Policy defines the metadata for an individual Policy within a Policy Pack.
+type Policy struct {
+	Name        string `json:"name"`
+	DisplayName string `json:"displayName"`
+
+	// Identifier is uniquely identified a Policy. It is a Policy Pack name and version as
+	// well as the Policy name encoded as a string in the following format:
+	// `<policyPackName>/<policyPackVersion>/<policyName>`
+	// For example, "aws-security/5/no-public-s3-buckets"
+	Identifier string `json:"identifier"`
```

Comment:
If this is a unique ID, who assigns it? The CLI? The policy pack author? If the latter, what happens if two policy packs take the same ID?

Reply:
this is a combination of unique information that has already been provided. I realize this is not needed under `Policy` as the server can generate it.  In the case of the `PolicyEvent`, the CLI will need to generate it from the provided PP/ PP version / name of Policy.